shell/platform/darwin/macos/framework/Source/FlutterAccessibilityMac.mm

```diff
@@ -0,0 +1,575 @@
+    return true;
+  }
+  return systemVersion.minorVersion >= target.minorVersion;
+}
+
+bool SystemVersionLessOrEqualTo(NSOperatingSystemVersion target) {
+  NSOperatingSystemVersion systemVersion = [[NSProcessInfo processInfo] operatingSystemVersion];
+  if (systemVersion.majorVersion < target.majorVersion) {
+    return true;
+  } else if (systemVersion.majorVersion > target.majorVersion) {
+    return false;
+  }
+  return systemVersion.minorVersion <= target.minorVersion;
+}
+
+// Native mac notifications fired.
+NSString* const NSAccessibilityAutocorrectionOccurredNotification = @"AXAutocorrectionOccurred";
+NSString* const NSAccessibilityLoadCompleteNotification = @"AXLoadComplete";
+NSString* const NSAccessibilityInvalidStatusChangedNotification = @"AXInvalidStatusChanged";
+NSString* const NSAccessibilityLiveRegionCreatedNotification = @"AXLiveRegionCreated";
+NSString* const NSAccessibilityLiveRegionChangedNotification = @"AXLiveRegionChanged";
+NSString* const NSAccessibilityExpandedChanged = @"AXExpandedChanged";
+NSString* const NSAccessibilityMenuItemSelectedNotification = @"AXMenuItemSelected";
+
+// Attributes used for NSAccessibilitySelectedTextChangedNotification and
+// NSAccessibilityValueChangedNotification.
+NSString* const NSAccessibilityTextStateChangeTypeKey = @"AXTextStateChangeType";
+NSString* const NSAccessibilityTextStateSyncKey = @"AXTextStateSync";
+NSString* const NSAccessibilityTextChangeElement = @"AXTextChangeElement";
+NSString* const NSAccessibilityTextEditType = @"AXTextEditType";
+NSString* const NSAccessibilityTextChangeValue = @"AXTextChangeValue";
+NSString* const NSAccessibilityChangeValueStartMarker = @"AXTextChangeValueStartMarker";
+NSString* const NSAccessibilityTextChangeValueLength = @"AXTextChangeValueLength";
+NSString* const NSAccessibilityTextChangeValues = @"AXTextChangeValues";
+
+enum AXTextStateChangeType {
+  AXTextStateChangeTypeUnknown,
+  AXTextStateChangeTypeEdit,
+  AXTextStateChangeTypeSelectionMove,
+  AXTextStateChangeTypeSelectionExtend
+};
+
+enum AXTextEditType {
+  AXTextEditTypeUnknown,
+  AXTextEditTypeDelete,
+  AXTextEditTypeInsert,
+  AXTextEditTypeTyping,
+  AXTextEditTypeDictation,
+  AXTextEditTypeCut,
+  AXTextEditTypePaste,
+  AXTextEditTypeAttributesChange
+};
+
+const int kLiveRegionChangeIntervalMS = 20;
+
+extern "C" {
+
+// The following are private accessibility APIs required for cursor navigation
+// and text selection. VoiceOver started relying on them in Mac OS X 10.11.
+
+AXTextMarkerRef AXTextMarkerCreate(CFAllocatorRef allocator, const UInt8* bytes, CFIndex length);
+
+}  // extern "C"
+
+// Static.
+FlutterAccessibility* FlutterAccessibility::Create() {
+  return new FlutterAccessibilityMac();
+}
+
+FlutterAccessibilityMac::FlutterAccessibilityMac() {
+  ax_platform_node_ = AXPlatformNode::Create(this);
+  FML_DCHECK(ax_platform_node_);
+}
+
+FlutterAccessibilityMac::~FlutterAccessibilityMac() {
+  ax_platform_node_->Destroy();
```

Comment:
If this was a smart pointer, this would be unnecessary, assuming its own destructor calls Destroy.